### PR TITLE
Chore: Docs site QOL updates

### PR DESCRIPTION
**Pull request by thomasdigby**
- Fixed issue with logo not linking to homepage
- Updated theme selection to persist to session storage
- Updated landing page and icons page to show more content on wider viewports
- Updated landing page component list to be fully alphabetised
- Tweaked icons design to fit better visually with the rest of the site (and make it more usable)

Screen Shot 2024-01-10 at 12 09 25
Screen Shot 2024-01-10 at 12 09 11

### CODE REVIEW

File: documentation/components/markdown/editor-component/Cards.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import {
   Grid,
   Box,
   Heading,
-  Stack,
```
Comment (andoulla) [inline documentation/components/markdown/editor-component/Cards.tsx:7]: Goodbye, old friend!

File: documentation/content/landing/index.md [hunk 2]
```diff
@@ -1,385 +1,8 @@
...
-      ee.svg\",\"heading\":\"Checkbox tree\",\"description\":\"Input with a tree
-      structure that allows multiple selections across various levels of a
-      hierarchy.\",\"link\":{\"href\":\"/components/form/primitives/checkbox-tr\
-      ee\"},\"tags\":[\"Inputs\",\"Filter\"]},{\"image\":\"/assets/images/tree.\
-      svg\",\"heading\":\"Tree\",\"description\":\"Represents data in a
-      hierarchical or tree-like structure, where items are organized into parent
-      and child
-      nodes.\",\"link\":{\"href\":\"/components/navigation/tree\"},\"tags\":[\"\
-      List and tables\"]}]} />"
+  - content: '<Cards showSearch={true} showTagsFilter={true}
```
Comment (andoulla) [inline documentation/content/landing/index.md:4]: not sure how to review this but I'm assuming it is the part where it makes the cards alphabetised? 

I have a question, do we need to manually add new components to this list?

**Review by andoulla — APPROVED** [review]

File: documentation/content/landing/index.md (hunk 2 above)
Comment (thomasdigby) [reply documentation/content/landing/index.md:4]: Yes this is the alphabetisin', and yes I believe we do have to manually add them. Do you think I should sort items as part of the `Cards` render? 🤔

Comment (andoulla) [reply documentation/content/landing/index.md:4]: btw I have already approved, I was just curious. This doesn't seem optimal but not a big enough issue to worry about it, unless you disagree Miguel

Comment (thomasdigby) [reply documentation/content/landing/index.md:4]: I'll leave as it is for now, we can add a sort in future when we add more components

File: documentation/components/markdown/editor-component/Cards.tsx (hunk 1 above)
Comment (thomasdigby) [reply documentation/components/markdown/editor-component/Cards.tsx:7]: 👋

**Review by LimeWub — APPROVED** [review]

**Review by MiguelGDLR — APPROVED** [review]

**Pull request merged**